### PR TITLE
Simplify Makefile

### PR DESCRIPTION
**Pull request by iredelmeier**
In particular, this change:

* Ensures that the race checker is always used for all tests
* Removes redundant commands
* Splits out `generate` and `lint` as their own commands
* Renames the `circle-ci` command to the more generic `ci`

### CODE REVIEW

**Comment by iredelmeier** [comment]
Trying to figure out what's causing the build failure. Maybe a cache issue?

File: Makefile [hunk 1]
```diff
@@ -7,10 +7,8 @@ ALL_DOCS := $(shell find . -name '*.md' -type f | sort)
 # All directories with go.mod files. Used in go mod tidy.
 ALL_GO_MOD_DIRS := $(shell find . -type f -name 'go.mod' -exec dirname {} \; | sort)
 
-GOTEST=go test
-GOTEST_OPT?=-v -timeout 30s
-GOTEST_OPT_WITH_RACE     = $(GOTEST_OPT) -race
-GOTEST_OPT_WITH_COVERAGE = $(GOTEST_OPT) -coverprofile=coverage.txt -covermode=atomic
+GOTEST=go test -v -timeout 30s -race
+GOTEST_WITH_COVERAGE = $(GOTEST) -coverprofile=coverage.txt -covermode=atomic
```
Comment (paivagustavo) [inline Makefile]: The usages of  `GOTEST_OPT_WITH_COVERAGE ` should be updated to `GOTEST_WITH_COVERAGE` that is probably the reason why it is failing on the CI.

Comment (jmacd) [reply Makefile]: Not sure. Why is there a timeout for coverage testing?

File: Makefile [hunk 2]
```diff
@@ -57,11 +36,11 @@ test-with-coverage:
 	    go tool cover -html=coverage.txt -o coverage.html); \
 	done
 
-.PHONY: circle-ci
-circle-ci: precommit test-clean-work-tree test-with-coverage test-race test-386 examples
+.PHONY: ci
+ci: precommit check-clean-work-tree test-with-coverage test-386
```
Comment (rghetia) [inline Makefile]: why is examples removed?

File: Makefile (hunk 1 above)
Comment (iredelmeier) [reply Makefile]: Thanks @paivagustavo!

@jmacd there always has been. Whether we want moving forward is a separate question 😂

File: Makefile (hunk 2 above)
Comment (iredelmeier) [reply Makefile]: @rghetia it's run by `precommit`.

Comment (rghetia) [reply Makefile]: SG. I missed that it was in precommit.

**Pull request merged**